addons/mail/static/src/new/core/message_model.js

```diff
@@ -219,4 +219,12 @@ export class Message {
     get failureNotifications() {
         return this.notifications.filter((notification) => notification.isFailure);
     }
+
+    get allowReplies() {
+        return (
+            this.originThread?.type !== "chatter" ||
+            (this._store.discuss.threadLocalId === this._store.discuss.inbox.localId &&
+                this._store.discuss.inbox.messageIds.includes(this.id))
+        );
+    }
```

Comment:
I don't think it's correct like this, the condition should depend on the view currently displayed, I don't think discuss loses its thread when a chatter is displayed.

Code should probably stay in component.